File: src/main/java/dev/paigewatson/layoutmaster/models/rollingstock/NullCarType.java
package dev.paigewatson.layoutmaster.models.rollingstock;

import dev.paigewatson.layoutmaster.data.CarTypeDAL;
import dev.paigewatson.layoutmaster.models.goods.GoodsType;

import java.util.Collections;

public class NullCarType extends BaseCarType
{
    public NullCarType()
    {
        this.aarDesignation = AARDesignation.NULL;
        this.carriedGoodsList = Collections.emptyList();
        this.id = "";
    }

    public boolean isNull()
    {
        return true;
    }


    public boolean canCarry(GoodsType goodsType)
    {
        return false;
    }

    public boolean isOfType(AARDesignation designation)
    {
        return false;
    }

    public AARDesignation displayName()
    {
        return AARDesignation.NULL;
    }

    public CarType saveToDatabase(CarTypeDAL carTypeDAL)
    {
        return null;
    }


    @Override
    public String toString()
    {
        return "NullCarType{" +
                "id=''" +
                ", aarDesignation=''" +
                ", carriedGoodsList=''" +
                "}";
    }

}

File: src/main/java/dev/paigewatson/layoutmaster/models/rollingstock/NullCarType.kt
package dev.paigewatson.layoutmaster.models.rollingstock

import dev.paigewatson.layoutmaster.data.CarTypeDAL
import dev.paigewatson.layoutmaster.models.goods.GoodsType

class NullCarType : BaseCarType() {


    override fun canCarry(goodsType: GoodsType): Boolean {
        return false
    }

    override fun isOfType(designation: AARDesignation): Boolean {
        return false
    }

    override fun displayName(): AARDesignation {
        return AARDesignation.NULL
    }

    override fun saveToDatabase(carTypeDAL: CarTypeDAL): CarType {
        return this
    }

    override val isNull: Boolean
        get() = true

    override fun toString(): String {
        return "NullCarType{" +
                "id=''" +
                ", aarDesignation=''" +
                ", carriedGoodsList=''" +
                "}"
    }

    init {
        aarDesignation = AARDesignation.NULL
        carriedGoodsList = emptyList()
        id = ""
    }
}
